tools/wt_cmp_uri

```diff
@@ -0,0 +1,308 @@
+            break
+        nrecords += 1
+        k1 = cc1.cursor.get_key()
+        v1 = cc1.cursor.get_value()
+        k2 = cc2.cursor.get_key()
+        v2 = cc2.cursor.get_value()
+        verbose('Version: GOT: {}:{} {}:{}'.format(k1,v1,k2,v2))
+        while k1 != k2:
+            if k1 < k2:
+                print('missing key={} on {}'.format(str(k1), cc2.name))
+                if cc1.cursor.next() != 0:
+                    print('EOF on {}'.format(cc1.name))
+                nrecords += 1
+                k1 = cc1.cursor.get_key()
+                v1 = cc1.cursor.get_value()
+            else:
+                print('missing key={} on {}'.format(str(k2), cc1.name))
+                if cc2.cursor.next() != 0:
+                    print('EOF on {}'.format(cc2.name))
+                nrecords += 1
+                k2 = cc2.cursor.get_key()
+                v2 = cc2.cursor.get_value()
+
+        # At this point, the keys are the same
+        if v1 != v2:
+            print('for key={}, value is different: {}: {}, {}: {}'.format(k1, cc1.name, v1, cc2.name, v2))
+        verbose('Version: finished compare')
+
+        ret1 = cc1.cursor.next()
+        ret2 = cc2.cursor.next()
+
+    if ret2 == 0:
+        print('EOF on {}'.format(cc1.name))
+    print('{} records read on version cursor'.format(nrecords))
+
+def get_dir_uri(name):
+    slash = name.rindex('/')
+    return (name[0:slash], name[slash+1:])
+
+args = sys.argv[1:]
+timestamp1 = None
+timestamp2 = None
+
+while len(args) > 1 and args[0][0] == '-':
+    if args[0] == '-v':
+        verboseFlag = True
+        args = args[1:]
+    elif args[0] == '-t':
+        timestamp1 = args[1]
+        timestamp2 = args[1]
+        args = args[2:]
+    else:
+        usage_exit()
+
+if len(args) < 2:
+    usage_exit()
+arg1 = args[0]
+(wtdir1, uri1) = get_dir_uri(arg1)
+args = args[1:]
+
+# Look for a second -t argument.
+while len(args) > 1 and args[0][0] == '-':
+    if args[0] == '-t':
+        timestamp2 = args[1]
+        args = args[2:]
+
+if len(args) != 1:
+    usage()
+arg2 = args[0]
+(wtdir2, uri2) = get_dir_uri(arg2)
+
+# We do allow wtdir1 and wtdir2 to be the same, it is useful to compare at different timestamps.
+# But when they are the same, we must use the same connection.
+
+verbose('wiredtiger_open({})'.format(wtdir1))
+conn1 = wiredtiger_open(wtdir1, 'readonly')
+if wtdir1 == wtdir2:
+    conn2 = conn1
+else:
+    verbose('wiredtiger_open({})'.format(wtdir2))
+    conn2 = wiredtiger_open(wtdir2, 'readonly')
+
+try:
+    cc1 = get_compare_cursor(conn1, timestamp1, uri1, arg1)
+except:
+    print('Failed opening {} in {} at timestamp {}'.format(uri1, wtdir1, timestamp1))
+    raise
+try:
+    cc2 = get_compare_cursor(conn2, timestamp2, uri2, arg2)
+except:
+    print('Failed opening {} in {} at timestamp {}'.format(uri2, wtdir2, timestamp2))
+    raise
+
+# Not running with version cursors, it's not quite ready for prime time.
+compare_cursors(cc1, cc2, False)
```

Comment:
It would be best if `compare_cursors` could return an error code (either directly or via an exception) and then the whole program sets the exit code based on whether it was a success or not.